### PR TITLE
20newsgroup

### PR DESCRIPTION
**Pull request by mariamabarham**
Add 20Newsgroup dataset.
#353

### CODE REVIEW

File: datasets/newsgroup/newsgroup.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import os
+
+import nlp
+
+
+_CITATION = """
```
Comment (lhoestq) [inline datasets/newsgroup/newsgroup.py:27]: Could you use the standardized way for citations, like in the other scripts ?

File: datasets/newsgroup/newsgroup.py [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    ,title = "Newsweeder: Learning to filter netnews"
+    ,year = 1995
+    ,booktitle = "Proceedings of the Twelfth International Conference on Machine Learning"
+    ,pages = "331-339"
+    }
+ """
+
+_DESCRIPTION = """
+The 20 Newsgroups data set is a collection of approximately 20,000 newsgroup documents, partitioned (nearly) evenly across 
+20 different newsgroups. To the best of my knowledge, it was originally collected by Ken Lang, probably for his Newsweeder:
```
Comment (lhoestq) [inline datasets/newsgroup/newsgroup.py:39]: whose "knowledge" ? ^^
Is there a description that is more factual ?

File: datasets/newsgroup/newsgroup.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        self.data_dir = data_dir
+
+
+class Newsgroups(nlp.GeneratorBasedBuilder):
+    
+    BUILDER_CONFIGS = [
+        NewsgroupConfig(
+            name=name,
+            description = _DESC[name.split('_')[0]],
+            data_dir=name.split('_')[1],
```
Comment (lhoestq) [inline datasets/newsgroup/newsgroup.py:110]: Maybe we can rename this variable ? We use `data_dir` as the directory for manually downloaded data.
Something like `sub_directory` for example ?

File: datasets/newsgroup/newsgroup.py (hunk 2 above)
Comment (mariamabarham) [reply datasets/newsgroup/newsgroup.py:39]: You're right.

**Pull request merged**